staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -478,7 +478,7 @@ func NewGenericServerResponse(code int, verb string, qualifiedResource schema.Gr
 // IsNotFound returns true if the specified error was created by NewNotFound.
 // It supports wrapped errors.
 func IsNotFound(err error) bool {
-	return ReasonForError(err) == metav1.StatusReasonNotFound
+	return ReasonForError(err) == metav1.StatusReasonNotFound || codeForError(err) == http.StatusNotFound
```

Comment:
Rather than adding a helper just to extract code, and type-extracting every error twice, could do both in a single step. Combined with the suggestion to only fall back to code for unrecognized reasons, could look like this:

```go
reason, code := ReasonAndCodeForError(err)
if reason == metav1.StatusReasonNotFound {
  return true
}
if !knownReasons[reason] && code == http.StatusNotFound {
  return true
}
return false
```

Reply:
that approach would also make the `IsConflict` change consistent... since `StatusReasonAlreadyExists` would not pass the `if !knownReasons[reason]` test